addons/pos_cache/models/pos_cache.py

```diff
@@ -49,7 +50,7 @@ def get_cache(self, domain, fields):
             self.product_fields = str(fields)
             self.refresh_cache()
 
-        return json.loads(base64.decodestring(self.cache))
+        return json.loads(pycompat.to_text(base64.decodestring(self.cache)))
```

Comment:
Only change I'd see is use `.decode('utf-8')` here rather than the more generic to_text, we know that the content ought be utf-8 encoded since that's what is used a few lines above.